src/components/Modal/Modal.tsx

```diff
@@ -271,14 +351,39 @@ class Modal extends React.PureComponent<ModalProps> {
   }
 
   render() {
-    const { className, isOpen, style, zIndex, ...rest } = this.props
+    const {
+      className,
+      isOpen,
+      kind,
+      state,
+      style,
+      version,
+      zIndex,
+      ...rest
+    } = this.props
+    const v2 = version === 2
 
     const componentClassName = classNames(
       'c-Modal',
+      v2 && 'v2',
       isOpen && 'is-open',
+      state === 'danger' && 'is-danger',
+      kind === MODAL_STYLES.DEFAULT && 'is-default',
+      kind === MODAL_STYLES.ALERT && 'is-alert',
+      kind === MODAL_STYLES.BRANDED && 'is-branded',
+      kind === MODAL_STYLES.SEQUENCE && 'is-sequence',
       className
     )
 
+    const innerWrapperClassName = classNames(
+      'c-Modal__innerWrapper',
+      v2 && 'v2',
+      kind === MODAL_STYLES.DEFAULT && 'is-default',
+      kind === MODAL_STYLES.ALERT && 'is-alert',
+      kind === MODAL_STYLES.BRANDED && 'is-branded',
+      kind === MODAL_STYLES.SEQUENCE && 'is-sequence'
```

Comment:
this pattern is repeated quite often. What do you think if we move that to a modal utility:

```javascript
export const getModalKindClassName = kind => {
  switch(kind){
    case MODAL_STYLES.ALERT: 
      return 'is-alert',
    case MODAL_STYLES.BRANDED: 
      return 'is-branded',
    case MODAL_STYLES.SEQUENCE: 
      return 'is-sequence',
    default:
      return 'is-default',
  }
}
```

Reply:
done and done. great feedback.